test/nats.test.ts

```diff
@@ -0,0 +1,14 @@
+import * as fc from '../src/index'
+import {it} from 'mocha'
+import {expect} from 'chai'
+
+describe('Nat tests', () => {
+  it('finds that the sum of two natural numbers is greater or equal than 0', () => {
+    expect(fc.scenario()
+      .forall('a', fc.nat(-10, 10))
+      .forall('b', fc.nat())
+      .then(({a, b}) => a + b >= 0)
```

Comment:
This is odd, isn't it? According to the specification, `a` will be bound to a `NoArbitrary` because `min < 0`. So what exactly is the `a` bound to in the expression `a + b`? Is this a [vacuous truth](https://en.wikipedia.org/wiki/Vacuous_truth) in propositional logic? Are we saying `forall a in empty set, then...`. If so, then this is tautologically true, and the test description is not saying what it's actually doing. In fact, I would argue this test *as is* belongs to other suite, as it is not speficially testing a behavior of `nat`'s, but a much more generic behavior.

Reply:
Yeah, I agree. When I said the test case for this bug was "stupid" I was thinking about what you have said now. In fact, at first I even developed such test in the `Arbitrary.test.ts` suite. Do you guys have any suggestions to what should be done?

Reply:
I'll let @arestivo and @ruippeixotog chime in, but yeah, this seems like something that should be testes on the `Arbitrary` level.

Reply:
Shouldn't it be "forall a < 0 and b > a then fc.nat(a, b) always returns NoArbitrary" Also a test for any a,b where a > b...

I also agree with Hugo's stance that NoArbitrary should be always false in existential checks.

Reply:
I'm proposing some new tests below. Please review :)

Reply:
Shouldn't invalid parameters return something else than a NoArbitrary? Wouldn't that eventually lead a developer to being tricked into thinking a test passes when in fact he is getting a NoArbitrary and doing a forall check that always returns true.

Reply:
That's what I meant when I said:

> It seems that if someone tries to instantiate a nat with min = -5 it gets one where min = 0 without so much as a warning. 

Right now we don't have a mechanism in place to raise errors or warnings. Ideally, this would be a constraint for the type system, but we are not there yet. I think it is reasonable to assume the behavior for _vacuous truth_, specially because they might arise during a shriking process. I'm still doubtful if this is the behavior we want from factory methods.

Reply:
I guess I have a different idea of what `NoArbitrary` means, but insofar as we're treating it as the null object and the empty generator, I strongly disagree with the choice of returning `NoArbitrary` for `min < 0`.

The formally correct answer for "all natural numbers between N and M" for N <= 0 and M > 0 is the interval [1...M]. IMO it's reasonable to accept any value for `min`, but if you want the API to be stricter we should just throw an exception instead of letting subsequent computations continue.

Reply:
I agree we are treating `NoArbitrary` as a null object / empty generator. I also agree that the most transparent way to keep the API strict is to do something that disallows subsequent computations. I'll change the code to throw an exception, and if a better idea comes along, please chime in :)

Reply:
Eventually we could have an InvalidArbitrary... But I'm really not sure if it is any better than throwing an exception.

Reply:
After some offline discussion with @ruippeixotog, we agreed that the following behavior for `nat` would be consistent with the current factories:

- For `max < 0` should result in a `NoArbitrary`
- For `min < 0`, the correct answer would be the interval [0.. max] (we are considering N0).
- Hence, for all intended purposes, `nat(x, y) <=> nat(max(0, x), y)`

Reply:
I agree. In fact all natural numbers between -3 and 3 are 0, 1, 2 and 3.